lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -382,17 +373,27 @@ def test_empty_query(
         index = self.build_index(
             server_config, self.date_range(self.payload["start"], self.payload["end"])
         )
-
-        response = query_api(
-            self.pbench_endpoint,
-            self.elastic_endpoint,
-            self.payload,
-            index,
-            expected_status,
-            headers=build_auth_header["header"],
-            status=HTTPStatus.OK,
-            json=self.empty_es_response_payload,
-        )
+        if self.api_method == API_METHOD.GET:
+            response = query_api_get(
+                self.pbench_endpoint,
+                self.elastic_endpoint,
+                index,
+                expected_status,
+                headers=build_auth_header["header"],
+                status=HTTPStatus.OK,
+                json=self.empty_es_response_payload,
+            )
+        elif self.api_method == API_METHOD.POST:
```

Comment:
I'm not really happy with the use of `elif`, here:  if we're reasonably sure that `self.api_method` is either `GET` or `POST`, then this `elif` should be a simple `else`; however, if we're making allowances for `self.api_method` to be something else (now or in the future), then there should be an `else` clause after line 396 which does something intelligent (like failing or skipping the test).

(You could also consider expanding the conditional at line 368.)